content/labs/Watsonx/WatsonxAI/files/203/LangChain_Memory.ipynb

```diff
@@ -1,304 +1 @@
act 3 factors that caused the customer to be unhappy.                             Put each factor on a new line. Customer complaint: I am writing you this statement to delete the                             following information on my credit report. The items I need deleted are listed in the report.                             I am a victim of identity thief, I demand that you remove these errors to correct my report immediately!                             I have reported this to the federal trade commission and have attached the federal trade commission affidavit.                             Now that I have given you the following information, please correct my credit report or I shall proceed with involving my attorney!                             Numbered list of complaints:', additional_kwargs={}, response_metadata={}), AIMessage(content=' 1. Credit report errors, 2. Identity theft, 3. Failure to correct errors in a timely manner.', additional_kwargs={}, response_metadata={}), HumanMessage(content='From the following customer complaint, extract 3 factors that caused the customer to be unhappy.                             Put each factor on a new line. Customer complaint: I am writing you this statement to delete the                             following information on my credit report. The items I need deleted are listed in the report.                             I am a victim of identity thief, I demand that you remove these errors to correct my report immediately!                             I have reported this to the federal trade commission and have attached the federal trade commission affidavit.                             Now that I have given you the following information, please correct my credit report or I shall proceed with involving my attorney!                             Numbered list of complaints:', additional_kwargs={}, response_metadata={}), AIMessage(content=' 1. Credit report errors, 2. Identity theft, 3. Failure to correct errors in a timely manner.', additional_kwargs={}, response_metadata={}), HumanMessage(content='Does the numbered list of complaints contain a statement about identity fraud? Provide a short answer: yes or no.', additional_kwargs={}, response_metadata={}), AIMessage(content='**\\n\\nYes', additional_kwargs={}, response_metadata={})])}\n"]}],"source":["# For debugging, print the history of the conversation\n","print(local_memory)"]},{"cell_type":"markdown","metadata":{"pycharm":{"name":"#%% md\n"}},"source":["### Authors: \n","- **Elena Lowery**, Data and AI Architect\n","- **Josefina Casanova**, Engagement Lead, Build Lab Americas. Edited for L4 watsonx course. 2024"]}],"metadata":{"kernelspec":{"display_name":"Python 3.11","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.11.9"}},"nbformat":4,"nbformat_minor":4}
```

Comment:
For memory we'll need another package: `!{sys.executable} -m pip install langchain-community | tail -n 1`

Reply:
Might want to look at formatting some of the output in this file too.

Reply:
Done, added package and clarified as with the other notebooks. 

As in output you mean clearing it for export, or that doesn't display right?